pipeline/auditors/uka.py

```diff
@@ -2,31 +2,17 @@
 
 from pipeline.auditors import BaseAuditor
 
+_is_one_digit = re.compile("^[0-9]$").fullmatch
 
-class UKAAuditor(BaseAuditor):
-    """A class used to warn for comprehensive classification, i.e. only one 1 digit classification"""
 
-    regexp_one_digit = re.compile("^[0-9]$")
+class UKAAuditor(BaseAuditor):
+    """Used to warn for comprehensive classification, i.e. only one 1 digit classification"""
 
     def __init__(self):
         self.name = UKAAuditor.__name__
 
     def audit(self, publication, audit_events, _harvest_cache, _session):
-        comprehensive = False
-        ukas = publication.ukas()
-        if self._only_1digits(ukas):
-            comprehensive = True
-            # logger.debug('Comprehensive classification found', extra={'auditor': self.name})
-        new_audit_events = self._add_audit_event(audit_events, comprehensive)
-        return publication, new_audit_events
-
-    def _add_audit_event(self, audit_events, comprehensive):
-        code = "UKA_comprehensive_check"
-        audit_events.add_event(self.name, code, comprehensive)
-        return audit_events
+        comprehensive = any(_is_one_digit(code) for code in codes)
```

Comment:
```suggestion
        comprehensive = any(_is_one_digit(code) for code in publication.ukas())
```
`codes` is undefined